src/borg/locking.py

```diff
@@ -170,15 +170,22 @@ def release(self):
         if not self.by_me():
             raise NotMyLock(self.path)
         os.unlink(self.unique_name)
-        try:
-            os.rmdir(self.path)
-        except OSError as err:
-            if err.errno not in (errno.ENOTEMPTY, errno.EEXIST, errno.ENOENT):
-                # EACCES or EIO or ... = we cannot operate anyway, so re-throw
-                raise err
-            # else:
-            # Directory is not empty or doesn't exist any more.
-            # this means we lost the race to somebody else -- which is ok.
+        retry = 0
+        while retry < 42:
+            retry += 1
```

Comment:
Is a `while` loop needed here? Or could `for` be used instead:
```py
for _ in range(42):
```

Reply:
Right. I started with `while True:` but then noticed that this might run into an endless loop in case the condition for breaking out of the loop is never met.